papis/commands/default.py

```diff
@@ -252,37 +258,52 @@ def run(verbose: bool,
     logging.basicConfig(level=getattr(logging, log), handlers=[handler])
     logger = logging.getLogger("default")
 
+    # }}}
+
+    # {{{ set up config
+
+    # NOTE: order of the configurations is intentional based on priority
+    #
+    #   3. global papis configuration file, e.g. `.config/papis/config`
+    #   2. library local configuration file, e.g. `LIB/.config`
+    #   1. command-line arguments, e.g. --set opentool firefox
+
+    # read in configuration file
     if config:
         papis.config.set_config_file(config)
         papis.config.reset_configuration()
 
-    for pair in set_list:
-        logger.debug("Setting '%s' to '%s'", *pair)
-        papis.config.set(pair[0], pair[1])
-
+    # read in configuration from current library
     if pick_lib:
-        _picked_libs = papis.pick.pick(papis.api.get_libraries())
-        if _picked_libs:
-            lib = _picked_libs[0]
+        picked_libs = papis.pick.pick(papis.api.get_libraries())
+        if picked_libs:
+            lib = picked_libs[0]
 
     papis.config.set_lib_from_name(lib)
     library = papis.config.get_lib()
 
     if not library.paths:
-        raise Exception("Library '{0}' does not have any existing folders"
-                        " attached to it, please define and create the paths"
-                        .format(lib))
-
+        raise RuntimeError(
+            "Library '{}' does not have any existing folders attached to it, "
+            "please define and create the paths"
+            .format(lib))
+
+    # Now the library should be set, let us check if there is a
+    # local configuration file there, and if there is one, then
+    # merge its contents
+    local_config_file = papis.config.getstring("local-config-file")
     for path in library.paths:
-        # Now the library should be set, let us check if there is a
-        # local configuration file there, and if there is one, then
-        # merge its contents
-        local_config_file = os.path.expanduser(
-            os.path.join(path,
-                         papis.config.getstring("local-config-file")))
+        local_config_file = os.path.expanduser(os.path.join(path, local_config_file))
```

Comment:
This only works for the first library but will fail for the others because you overwrite the variable.

Reply:
Nice catch! It's renamed in
https://github.com/papis/papis/compare/5827098cc978e5de394e168a7e60446eb6effaba..e2d5d5b65b3754300d1ce50206ea4c72ccedf747